alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -49,7 +51,39 @@ void RunnableObject::start(RunningPlan* rp)
     }
     ++_signalState;
     _signalContext.store(rp);
-    _engine->editScheduler().schedule(std::bind(&RunnableObject::doInit, this));
+    if(rp->getParent() && rp->getParent()->getBasicPlan()) {
```

Comment:
Running plan should pass parents blackboard reference (if this one doesn't require parameter) in start api

Reply:
I don't think this is worth it.  we need to look up the parent to get the appropriate plan attachment anyway if we do require parameters, and we would be looking it up in the same way a few methods up (running plan does not get a direct reference to the parent, and passes it through PlanPool before reaching here)